src/pip/_internal/utils/glibc.py

```diff
@@ -14,6 +14,19 @@ def glibc_version_string():
     # type: () -> Optional[str]
     "Returns glibc version string, or None if not using glibc."
 
+    try:  # https://github.com/pypa/pip/issues/6675#issue-463147612
+        glibc_version = os.confstr("CS_GNU_LIBC_VERSION").split()
+    except (AttributeError, OSError, ValueError):
+        pass  # os.confstr() or CS_GNU_LIBC_VERSION not available...
+    else:
+        if len(glibc_version) == 2:
+            return glibc_version[-1]
+
+    try:
```

Comment:
Putting the ctypes-based strategy and the new one in their own functions would make them easier to unit test.

Reply:
Yes, and there should _be_ tests of the new logic.